recirq/quantum_chess/quantum_board.py

```diff
@@ -167,19 +167,40 @@ def record_time(self, action: str, t0: float, t1: Optional[float] = None) -> Non
         self.debug_log += (f"{action} takes {t1 - t0:0.4f} seconds.\n")
         self.timing_stats[action].append(t1 - t0)
 
-    def sample_with_ancilla(self, num_samples: int
+    def suggest_num_reps(self, sample_size: int) -> int:
+        """Guess the number of raw samples needed to get sample_size results.
+
+        Assume that each post-selection is about 50/50.
+        Noise and error mitigation will discard reps, so increase the total
+        number of repetitions to compensate.
+        """
+        if len(self.post_selection) > 1:
+            sample_size *= 2 ** (len(self.post_selection) + 1)
+        if self.error_mitigation == enums.ErrorMitigation.Correct:
+            sample_size *= 2
+        if self.noise_mitigation > 0:
+            sample_size *= 3
+        if sample_size < 100:
+            sample_size = 100
+        return sample_size
+
+    def sample_with_ancilla(self, num_samples: int, num_reps: int = -1
```

Comment:
Why default of "-1"?  Why not set to Optional[int] = None, which would be more clear?

Reply:
OK I changed it.